src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -682,6 +720,67 @@ private int decodePublish(
         return progress;
     }
 
+    private int onDecodePublishSuccess(
```

Comment:
This method is just part of the `decodePublish` method and can be inlined, right?
If the method is too long, then let's instead find a way to hoist the property decoding into a separate visitor to simplify the loop logic in `decodePublish`.